app/models/session_type.rb

```diff
@@ -1,11 +1,13 @@
 class SessionType < ActiveRecord::Base
   belongs_to :event
   has_many :sessions
+  has_many :proposals
 
   validates_presence_of :name, :event
   validates_uniqueness_of :name, scope: :event
 
   scope :sort_by_name, ->{ order(:name) }
+  scope :public_types, ->{ where(public: true)}
```

Comment:
Drop _types on the scope.  With scopes it's rare to repeat the object name since you're chaining them.

Here's an example that is traditional in Rails:
event.session_types.public